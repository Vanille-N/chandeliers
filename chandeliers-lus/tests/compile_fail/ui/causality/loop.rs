//! All-in-one cyclic definitions.
chandeliers_lus::decl! {
    const X: int = X + 1;
}

chandeliers_lus::decl! {
    node foo() returns (i : int);
    let
        i = foo();
    tel
}

chandeliers_lus::decl! {
    node bar() returns (f : float);
    let
        f = 1.0 + f * 0.5;
    tel;
}

chandeliers_lus::decl! {
    node foo() returns ();
    let () = bar(); tel;

    node bar() returns ();
    let () = foo(); tel;
}

fn main() {}
