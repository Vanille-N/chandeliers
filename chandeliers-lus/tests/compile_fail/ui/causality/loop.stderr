error: Global `X` depends on itself
 --> tests/compile_fail/ui/causality/loop.rs:2:5
  |
2 |     const X: int = X + 1;
  |     ^^^^^^^^^^^^^^^^^^^^
  |
note: used here within its own definition
 --> tests/compile_fail/ui/causality/loop.rs:2:20
  |
2 |     const X: int = X + 1;
  |                    ^

error: Node `foo` depends on itself
 --> tests/compile_fail/ui/causality/loop.rs:6:5
  |
6 | /     node foo() returns (i : int);
7 | |     let
8 | |         i = foo();
9 | |     tel
  | |_______^
  |
note: used here within its own definition
 --> tests/compile_fail/ui/causality/loop.rs:8:13
  |
8 |         i = foo();
  |             ^^^

error: Variable `f` depends on itself
  --> tests/compile_fail/ui/causality/loop.rs:15:9
   |
15 |         f = 1.0 + f * 0.5;
   |         ^^^^^^^^^^^^^^^^^
   |
note: used here within its own definition
  --> tests/compile_fail/ui/causality/loop.rs:15:19
   |
15 |         f = 1.0 + f * 0.5;
   |                   ^

error: Node `bar` was found to be part of a dependency cycle
  --> tests/compile_fail/ui/causality/loop.rs:23:10
   |
23 |     node bar() returns ();
   |          ^^^
   |
note: The cycle also goes through Node `foo`
  --> tests/compile_fail/ui/causality/loop.rs:20:10
   |
20 |     node foo() returns ();
   |          ^^^
